### PR TITLE
Inject environment merged from all sources into the string expansion …

### PR DESCRIPTION
**Pull request by andreeis**
https://github.com/microsoft/vscode-cmake-tools/issues/1250 is not fixed with 1.7.
I either had a loophole in testing the original PR that targeted the issue or it regressed with more recent PRs.
This fix is ensuring an up to date environment is sent every time we expand a string.

### CODE REVIEW

File: src/drivers/driver.ts [hunk 1]
```diff
@@ -549,6 +549,7 @@ export abstract class CMakeDriver implements vscode.Disposable {
     return this.doRefreshExpansions(async () => {
       log.debug('Run _refreshExpansions cb');
       const opts = this.expansionOptions;
+      opts.envOverride = await this.getConfigureEnvironment();
```
Comment (bobbrow) [inline src/drivers/driver.ts:552]: Should `.envOverride` be set as part of `this.expansionOptions`?  I think that's a bigger change but seems like if we do it here we should be doing it everywhere else too.

Comment (andreeis) [reply src/drivers/driver.ts:552]: I can't quickly answer this. Do you want to look at this more thoroughly for 1.8 and move the bug out of 1.7.1?  I am thinking we can also include this fix as is for 1.7.1, to unblock the main scenario we are aware of and revisit for 1.8 whether there is a better place to do this check?

Comment (bobbrow) [reply src/drivers/driver.ts:552]: That proposal sounds good to me.  I'll approve this and then let's make sure we're tracking the full fix for 1.8.

**Review by bobbrow — APPROVED** [review]

File: src/drivers/driver.ts (hunk 1 above)
Comment (andreeis) [reply src/drivers/driver.ts:552]: https://github.com/microsoft/vscode-cmake-tools/issues/1798

**Pull request merged**